src/System.Net.Http/src/System/Net/Http/MultipartContent.cs

```diff
@@ -177,122 +172,57 @@ Collections.IEnumerator Collections.IEnumerable.GetEnumerator()
         // write "--" + boundary + "--"
         // Can't be canceled directly by the user.  If the overall request is canceled 
         // then the stream will be closed an exception thrown.
-        protected override Task SerializeToStreamAsync(Stream stream, TransportContext context)
+        protected override async Task SerializeToStreamAsync(Stream stream, TransportContext context)
         {
             Debug.Assert(stream != null);
-            Debug.Assert(_outputStream == null, "Operation already in progress");
-            Debug.Assert(_tcs == null, "Operation already in progress");
-            Debug.Assert(_nextContentIndex == 0, "Operation already in progress");
-
-            // Keep a local copy in case the operation completes and cleans up synchronously.
-            TaskCompletionSource<Object> localTcs = new TaskCompletionSource<Object>();
-            _tcs = localTcs;
-            _outputStream = stream;
-            _nextContentIndex = 0;
-
-            // Start Boundary, chain everything else.
-            EncodeStringToStreamAsync(_outputStream, "--" + _boundary + CrLf)
-                .ContinueWithStandard(WriteNextContentHeadersAsync);
-
-            return localTcs.Task;
-        }
-
-        private void WriteNextContentHeadersAsync(Task task)
-        {
-            if (task.IsFaulted)
-            {
-                HandleAsyncException("WriteNextContentHeadersAsync", task.Exception.GetBaseException());
-                return;
-            }
-
             try
             {
-                // Base case, no more content, finish.
-                if (_nextContentIndex >= _nestedContent.Count)
-                {
-                    WriteTerminatingBoundaryAsync();
-                    return;
-                }
+                // Write start boundary
```

Comment:
nit: add period at end of comment.  since you touched the file and/or added comments, please fix up rest of file.